### ISSUE TITLE
MASC crit affects supercharger

### ISSUE THREAD

**Issue by AlphaSled** [MegaMek/megamek#2563]
v.47.16

The bug occurs on turn six of this save game. The Thresher II fails its MASC roll, but passes its Supercharger check. The unit, however, suffered a critical hit to its Supercharger all the same.

[superchargerbug.sav.gz](https://github.com/MegaMek/megamek/files/5793972/superchargerbug.sav.gz)

**Comment by NickAragua** [comment]
Yep, Server.java, line 9707 - it fries the supercharger when MASC fails, because the MASC failure check doesn't distinguish between fried MASC and supercharger.

**Comment by pakfront** [comment]
I can take this along with 2749. They are intertwined

**Comment by TenkawaBC** [comment]
Fixed by https://github.com/MegaMek/megamek/pull/3434 . should have been closed with 2749.
